string source
float32 control_left
float32 control_right
